refinery/core/serializers.py

```diff
@@ -165,3 +166,43 @@ class Meta:
             'ready_for_igv_detail_view',
             'file_uuid'
         ]
+
+
+class _StrObjectField(serializers.Field):
+    def get_attribute(self, obj):
+        # Without this, it tries to access a field by the same name.
+        return obj
+
+    def to_representation(self, obj):
+        return str(obj)
+
+
+class _DeserializeJSONField(serializers.Field):
+
+    def to_representation(self, obj):
+        return json.loads(obj)
+
+
+class EventSerializer(serializers.ModelSerializer):
+    data_set = serializers.SlugRelatedField(
+        read_only=True,
+        slug_field='uuid'
+    )
+
+    user = serializers.SlugRelatedField(
+        read_only=True,
+        slug_field='username'
+    )
+
+    message = _StrObjectField()
+
+    json = _DeserializeJSONField()
```

Comment:
Is the standard [JSONField](http://www.django-rest-framework.org/api-guide/fields/#jsonfield) not sufficient here?

Reply:
You can use `JSONField` while specifying a `source` that points to a method on the `Event model that will spit out `Event.json` as a dict

See an example here: https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/tool_manager/serializers.py#L79